### PR TITLE
Fix class casting issue when building authentication sequence of portal apps

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Proposed changes in this pull request

$subject

#### Related Issues
- https://github.com/wso2/product-is/issues/20321

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java [hunk 1]
```diff
@@ -1250,22 +1252,22 @@ private void removeOrganizationSsoStepsForPortalApps(SequenceConfig effectiveSeq
         stepConfig.setAuthenticatorList(authenticatorList);
 
         // Remove the organization SSO authenticator from the graph nodes if already configured.
-        if (effectiveSequence.getAuthenticationGraph() == null ||
-                effectiveSequence.getAuthenticationGraph().getStartNode() == null) {
+        AuthenticationGraph authenticationGraph = effectiveSequence.getAuthenticationGraph();
+        if (authenticationGraph == null || authenticationGraph.getStartNode() == null) {
```
Comment (SujanSanjula96) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java:1256]: shall we assign `authenticationGraph.getStartNode()` to a variable and reuse

Comment (sadilchamishka) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java:1256]: It is taken into a variable in this line https://github.com/wso2/carbon-identity-framework/pull/5671/files#diff-e516f80a933232535c74b8fef4518f56b22514cd7aac461cc69fb6e5623319e8R1260
 Before that, the object is not yet casted.

Comment (SujanSanjula96) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java:1255]: I mean `AuthGraphNode authGraphNode = authenticationGraph.getStartNode();` and reuse in the three places

**Review by SujanSanjula96 — APPROVED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java (hunk 1 above)
Comment (sadilchamishka) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java:1255]: Yeah, got it.

**Review by SujanSanjula96 — DISMISSED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java (hunk 1 above)
Comment (sadilchamishka) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java]: Fixed by - https://github.com/wso2/carbon-identity-framework/pull/5671/commits/5297a9e38945234f1d9a206fcbc9ee378caaec20

**Review by SujanSanjula96 — APPROVED** [review]

**Pull request merged**